cmd/run.go

```diff
@@ -154,6 +204,22 @@ func (r *Runner) Run(ctx context.Context, patterns []string) error {
 		return err
 	}
 
+	var fn func(context.Context, *cmd.Run) error
```

Comment:
No disagreement with extracting switch case content into methods, but why move the switch up here? If it's all the same, can just leave switch where it is and make the actual invocations in each case